flask/wrappers.py

```diff
@@ -175,7 +174,10 @@ def on_json_loading_failed(self, e):
 
         .. versionadded:: 0.8
         """
-        raise BadRequest()
+        if current_app.config['DEBUG']:
```

Comment:
I mean that some user might e.g. want to use Flask's extended request objects without the rest of Flask... take a look at `max_content_length` where similar precautions are made.